python/ray/exceptions.py

```diff
@@ -263,14 +263,16 @@ class RayActorError(RayError):
             but it is there as a safety check.
     """
 
-    def __init__(self, cause: Union[RayTaskError, ActorDiedErrorContext] = None):
+    def __init__(self, cause: Union[RayTaskError, ActorDiedErrorContext, str] = None):
```

Comment:
Good catch. But it's better to fix it at this line https://github.com/ray-project/ray/blob/master/python/ray/_raylet.pyx#L778 

by passing an ActorDiedErrorContext instead of adding a new type.

Reply:
My concern with doing this is that the `RayActorError` will not have `actor_init_failed=True`.

I tried updating this function to return a `RayTaskError` instead, but then **only the traceback is only printed in the str() call**, not the `cause`.

I could do something hacky like:
```
raise RayActorError.from_task_error(
    RayTaskError(
        function_name=f"{class_name}.__init__",
        traceback_str=f"Failed to create the actor {core_worker}. "
                        "The failure reason is that you set the async flag, "
                        "but the actor has no any coroutine function.",
        cause=""
    )
)
```

Reply:
I don't think we should set `actor_init_failed=True` because this flag indicates that the actor failed when executing `__init__`.
For our case, the actor should fail before executing `__init__`.

CC @rkooo567 for double check.

Reply:
Oh! Thanks for the clarification @jovany-wang! Will move this to `ActorDiedErrorContext`